scripts/modules/VBFReweightModules.py

```diff
@@ -0,0 +1,192 @@
+        self.CV  = CV
+        self.C2V = C2V
+        self.kl  = kl
+        self.s1  = s1
+        self.s2  = s2
+        self.s3  = s3
+        self.s4  = s4
+        self.s5  = s5
+        self.s6  = s6
+
+        # Automatically build the matrix of the initial 6-equation system given the list of samples
+        M = build_matrix(sample_list)
+
+        # Declare the vectors of samples, couplings and components
+        # The vector of couplings
+        C = Matrix([
+            [CV**2 * kl**2] ,
+            [CV**4]         ,
+            [C2V**2]        ,
+            [CV**3 * kl]    ,
+            [CV * C2V * kl] ,
+            [CV**2 * C2V]   
+        ])
+
+        # The vector of components
+        V = Matrix([
+            [a]   ,
+            [b]   ,
+            [c]   ,
+            [iab] ,
+            [iac] ,
+            [ibc]  
+        ])
+
+        # The vector of samples (i.e. cross sections of the input samples)
+        S = Matrix([
+            [s1] ,
+            [s2] ,
+            [s3] ,
+            [s4] ,
+            [s5] ,
+            [s6] 
+        ])
+
+        # Invert the system matrix to get the resolution
+        Minv   = M.inv()
+
+        # Multiply the Minv for the couplings to get the coefficients of the six addendums of the final cross sections
+        coeffs = C.transpose() * Minv
+        self.coefficients = coeffs
+
+        # Final cross section from multiplying the coefficients for the samples
+        sigma  = coeffs*S
+        self.modeled_cross_section = sigma
+
+
+    # Returns the modeled cross section and histogram given the target couplings
+    def modelSignal(self,t_cv,t_c2v,t_kl,target_xs):
+
+        # Eval the final cross section
+        total_xs = self.modeled_cross_section[0].evalf(subs={
+            self.CV  : t_cv,
+            self.C2V : t_c2v,
+            self.kl  : t_kl,
+            self.s1  : self.sample_list[0].val_xs,
+            self.s2  : self.sample_list[1].val_xs,
+            self.s3  : self.sample_list[2].val_xs,
+            self.s4  : self.sample_list[3].val_xs,
+            self.s5  : self.sample_list[4].val_xs,
+            self.s6  : self.sample_list[5].val_xs,
+        })
+
+        # Eval the single coefficients
+        eval_coeffs = []
+        for coeff in self.coefficients:
+            eval_coeffs.append(coeff.evalf(subs={self.CV:t_cv, self.C2V:t_c2v, self.kl:t_kl}))
+
+        # Get a clone of the input histograms to be scaled and summed
+        hists = []
+        for sample in self.sample_list:
+            hists.append(sample.histo.Clone(sample.histo.GetName()))
+
+        # Normalize each histo to his coefficient * xs
+        for i,sample in enumerate(self.sample_list):
+            if target_xs == -1:
+                scaleTo(hists[i], eval_coeffs[i])
+            else:
+                scaleTo(hists[i], eval_coeffs[i] * target_xs / sample.val_xs)
```

Comment:
@camendola can you confirm that this normalization is correct?

Reply:
yes that's correct!